### PR TITLE
feature(build) - Add configuration for eslint and new task structure.

### PR DESCRIPTION
**Pull request by mcacek**
- Add a basic eslint configuration. There are lots of errors right now.
- Add initial structure for external/shared task configuration.

### CODE REVIEW

File: src/js/debounce.js [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-      // then we reset the timeout to wait for the difference
-      if (last < wait) {
-        timeout = setTimeout(later, wait - last);
-
-        // or if not we can null out the timer and run the latest
-      } else {
-        timeout = null;
-        func.apply(context, args);
-      }
+    // we need to save these in the closure
```
Comment (mcacek) [inline src/js/debounce.js]: autoindented...

File: src/gulp/tasks/scripts.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+var gulp = require('gulp');
+var eslint = require('gulp-eslint');
+var gulpConfig = require('../gulp-config');
+var taskConfig = gulpConfig.tasks.scripts;
+var eslintConfig = taskConfig.eslint;
+
+gulp.task('scripts:lint', scriptsLint);
+
+function scriptsLint() {
```
Comment (mcacek) [inline src/gulp/tasks/scripts.js]: Nothing depends on this due to lots of errors. This is proof of concept for gulp tasks and configuration.

File: src/gulp/tasks/images.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+var gulp = require('gulp');
+var config = require('../gulp-config');
+var taskConfig = config.tasks.images;
+var plumber = require('gulp-plumber');
+
+gulp.task('images', function(){
```
Comment (mcacek) [inline src/gulp/tasks/images.js]: This is the same task that existed before but is now a separate file with a dependency on the shared configuration.

File: src/gulp/gulp-config.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+var gulpConfig = {
+    paths: {
+        dist: './dist'
+    }
+};
+
+var tasks = require('./config/tasks-config')(gulpConfig);
```
Comment (mcacek) [inline src/gulp/gulp-config.js]: Tasks require this gulp config which generates a configuration that has all of the separate task configurations. This allows all task configuration to be organized but shareable.

File: src/gulp/config/tasks-config.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+module.exports = function(gulpConfig) {
+    return {
```
Comment (mcacek) [inline src/gulp/config/tasks-config.js]: I'll probably break this into separate files and have this file merge it all together but for now this is how it works. Establishing a standard approach to configuration naming and structure.

File: src/gulp/gulp-config.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+var gulpConfig = {
```
Comment (mcacek) [inline src/gulp/gulp-config.js]: This initial gulp config object is the shareable "common" configuration used by all tasks. Anything that is common to all tasks should go here.

**Comment by mcacek** [comment]
Lots more to do but this is a starting point. Let me know if any of this seems weird. I've been dealing with confusing build systems lately so I could be confused.  The configuration bit is based a little bit on how linemanjs does things. http://www.linemanjs.com/

File: package.json [hunk 7]
```diff
@@ -37,5 +37,10 @@
   "bugs": {
     "url": "https://github.com/aaronzirbes/react-lakitu/issues"
   },
-  "homepage": "https://github.com/aaronzirbes/react-lakitu#readme"
+  "homepage": "https://github.com/aaronzirbes/react-lakitu#readme",
+  "browserify": {
+    "transform": [
+      "reactify"
```
Comment (aaronzirbes) [inline package.json]: what's this do?

Comment (mcacek) [inline package.json]: This was each browserify gulp task before, but it's more flexible to have this in the package.json.

**Pull request merged**